src/components/Config/Config.tsx

```diff
@@ -70,15 +67,27 @@ function Config() {
   };
 
   const handleConnect = useCallback(async () => {
-    checkMetamaskInstalled();
+    metamaskState.connect();
     if (network !== Network.none) {
       setSubmitted(true);
       setCompleted(false);
     }
-  }, [network]);
+  }, [network, metamaskState]);
+
+  const handleAlreadyConnected = useCallback(async () => {
+    metamaskState.connect();
+    const connection = await connect();
+    if (connection.connectionParams.network === network) {
```

Comment:
This logic here can be simplified when you pass in the desired network - you don't need to check the network here